### ISSUE TITLE
Symbol grip with actor is displayed as "Object {  }"

### ISSUE THREAD

**Issue by Loirooriol** [devtools-html/devtools-core#842]
[Bug 1424722](https://bugzilla.mozilla.org/show_bug.cgi?id=1424722) will create actors for symbols so that they can be stored as variables.

The problem is that reps seems to consider that a grip with an actor must refer to an object, and displays `Object {  }` instead of `Symbol(name)`. The old console frontend works fine.

Cc @nchevobbe

**Comment by AnshulMalik** [comment]
I can look into this :)

**Comment by AnshulMalik** [comment]
I need some more inputs here please, Reps detects Symbol() and displays it as Symbol.
I might not be getting the point :(

**Comment by Loirooriol** [comment]
You need to add the patch from the linked bug, https://reviewboard.mozilla.org/r/207016/diff/1#index_header. It adds an actor to the symbol grip. Right now the symbol grip has no actor so it's displayed correctly.

**Comment by nchevobbe** [comment]
Anshul, this means we have to update the symbol stubs in reps code to match
what we'll have with oriol's patch. Then, we can see the test failing and
fix them.

My feeling is that here we probably default to object because supportObject
now return false.

**Comment by Loirooriol** [comment]
OK, the problem is that `getGripType` is supposed to return `"symbol"` for symbol grips.

```js
function getGripType(object, noGrip) {
  let type = typeof object;
  if (type == "object" && object instanceof String) {
    type = "string";
  } else if (object && type == "object" && object.type && noGrip !== true) {
    type = object.type;
  }

  if (isGrip(object)) {
    type = object.class;
  }

  return type;
}
```

But `isGrip` returns true is there is an `actor` property:

```js
function isGrip(object) {
  return object && object.actor;
}
```

so `object.class` is returned instead, which is undefined.

As I said in some other bug, this `isGrip` function is too naive and unreliable, it should just be removed in favor of the `noGrip` parameter.

And can grips really be string objects?

I guess it should simply be something like

```js
function getGripType(object, noGrip) {
  if (!object || noGrip) {
    return typeof object;
  }
  return object.type || object.class;
}
```

**Comment by nchevobbe** [comment]
@AnshulMalik do you have some time to look at this ? I think we should move quickly here since https://bugzilla.mozilla.org/show_bug.cgi?id=1424721 landed, which means Symbol will be broken in Nightly in the next days.

**Comment by AnshulMalik** [comment]
I am kind of lost here @nchevobbe, trying to make sense of the code.

**Comment by Loirooriol** [comment]
In fact [bug 1424721](https://bugzilla.mozilla.org/show_bug.cgi?id=1424721) is not problematic, but I would like this to be fixed so that I can fix bug [1424722](https://bugzilla.mozilla.org/show_bug.cgi?id=1424722).

@AnshulMalik Currently symbol grips are like

```js
({
  type: "symbol",
  name: "foo",
})
```

They will become something like

```js
({
  type: "symbol",
  name: "foo",
  actor: "server1.conn1.child1/symbol123",
})
```

So `getGripType` needs to be updated to what I said in https://github.com/devtools-html/devtools-core/issues/842#issuecomment-351102203.

**Comment by AnshulMalik** [comment]
Yeah, thanks @nchevobbe 

I agree with @Loirooriol, `isGrip` is no longer needed.
